ibis/expr/operations/core.py

```diff
@@ -95,37 +93,6 @@ def _pp(x):
 
         return '{}({})'.format(opname, ', '.join(pprint_args))
 
-    def __getstate__(self) -> Dict[str, AnyT]:
-        """The attributes _expr_cached and _hash are
-        used as caches; they can be excluded from
-        serialization without affecting correctness.
-
-        Excluding _expr_cached and _hash from serialization
-        will allow the serialized bytes to be the same for
-        equivalent Node objets.
```

Comment:
Do these comments still apply?

Reply:
We only serialize the fields part of the signature, nothing else.

Reply:
I think this is probably a regression then, because those fields are explicitly being serialized.

Reply:
The behavior hasn't changed, `_expr_cached` and `_hash` are not part of the signature https://github.com/ibis-project/ibis/pull/3105/files#diff-09f7d95a14c73bb445f8c811277d0b3be65436979662a505a2275e6059ae5237R160